File: Toolkit/Main/src/main/java/com/redelf/commons/data/Storage.java
package com.redelf.commons.data;

import com.redelf.commons.management.DataManagement;

public class Storage {

    public static <T> boolean put(String key, T value) {

        return DataManagement.STORAGE.push(key, value);
    }

    public static <T> T get(String key) {

        return DataManagement.STORAGE.pull(key);
    }

    public static <T> T get(String key, T defaultValue) {

        final T res =  DataManagement.STORAGE.pull(key) ;
        return res == null ? defaultValue : res;
    }

    public static boolean delete(String key) {

        return DataManagement.STORAGE.delete(key);
    }

    public static boolean deleteAll() {

        return DataManagement.STORAGE.erase();
    }

    public static boolean contains(String key) {

        return DataManagement.STORAGE.contains(key);
    }
}

File: Toolkit/Main/src/main/java/com/redelf/commons/persistance/base/Storage.kt
package com.redelf.commons.persistance.base

import com.redelf.commons.lifecycle.InitializationWithContext
import com.redelf.commons.lifecycle.ShutdownSynchronized
import com.redelf.commons.lifecycle.TerminationSynchronized

/*
    FIXME: We do not need both ShutdownSynchronized and TerminationSynchronized.
       Check other the files as well!
* */
interface Storage<T> : ShutdownSynchronized, TerminationSynchronized, InitializationWithContext {

    fun put(key: String?, value: T): Boolean

    fun get(key: String?): T

    fun delete(key: String?): Boolean

    fun deleteAll(): Boolean

    fun count(): Long

    fun contains(key: String?): Boolean
}
